pkg/controller/machinehealthcheck/machinehealthcheck_controller.go

```diff
@@ -392,8 +435,23 @@ func (t *target) remediationStrategyReboot(r *ReconcileMachineHealthCheck) error
 	glog.Infof("Machine %s has been unhealthy for too long, adding reboot annotation", t.Machine.Name)
 	t.Node.Annotations[machineRebootAnnotationKey] = ""
 	if err := r.client.Update(context.TODO(), t.Node); err != nil {
+		r.recorder.Eventf(
```

Comment:
You can get here with nil `nodeRef`. If the nodeRef is never set you will panic here. You should really be operating over the machine instead.

Reply:
Good point. Changed to use Machine.